lib/puppet/configurer.rb

```diff
@@ -44,6 +44,12 @@ def init_storage
     end
   end
 
+  def init_http_pool_monitoring(report, http_pool)
+    if http_pool.is_a? Puppet::Network::HTTP::Pool
+      http_pool.report = report
```

Comment:
It would probably be better to push the report onto the context stack, and then access the report using `Puppet.lookup(:report)` from within the http pool.